R/format_value.R

```diff
@@ -300,8 +300,8 @@ format_value <- function(x, format = NULL, output = c("ascii", "html"), na_str =
         format, " are of different length"
       )
     }
-    if (length(na_str) < length(x)) {
-      na_str <- rep(na_str, length.out = length(x))
+    if (length(na_str) < sum(is.na(x))) {
```

Comment:
Ok, then adding a vector is also possible. Do you know if this is documented? I am investigating

Reply:
`format_value(c(1, NA, NA), "xx", na_str = c("as", "sa"))` This seems to be undocumented

Reply:
<img width="961" alt="Screenshot 2024-01-22 102433" src="https://github.com/insightsengineering/formatters/assets/11279768/5aab36f5-4d8d-426a-928c-3e60117a3a96">
Let's limit this. Can I ask you to merge the formatters `@param na_str` into one doc entry that specifies that you can insert a vector of the correct length? (I also suspect this has no tests or checks for the right match)

Reply:
@Melkiades I don't think the other two functions with `na_str` parameters need to be able to take vectors as `na_str` (unless we want to fully refactor `round_fmt` to accept vectors arguments for `digits`/`na_str`, but there doesn't seem to be a use case as of yet for this)

Reply:
@Melkiades on second thought, I think we should make a separate PR to tackle this issue - we have to also consider the fact that users can supply multiple `na_str` values for multi-value formats (e.g. `na_str = c("lo", "hi")` for `"xx.xx - xx.xx"`), which would complicate enabling vector input.

Reply:
I see! Have you made it already? I think we need to understand how it is exactly applied and uniform that. In other words, we have to see exactly what providing a vector entails. From a user perspective, you would do 3 things: (1) one for all na_str (current default). (2) 2+ for specific formats (how do we select them externally? I do not remember, is na_str working as formats, that you can specify different values with named vectors?) (3) assign them all in order (if less, cycle)